File: frontend/src/app/core/api/api.routes.ts
const API_BASE = '/api';

export const API_ROUTES = {
  auth: {
    login: `${API_BASE}/auth/login`,
    refresh: `${API_BASE}/auth/refresh`,
    logout: `${API_BASE}/auth/logout`,
  },
  conversations: {
    root: `${API_BASE}/conversations`,
    messages: (conversationId: string) =>
      `${API_BASE}/conversations/${conversationId}/messages`,
    streamMessage: (conversationId: string) =>
      `${API_BASE}/conversations/${conversationId}/messages/stream`,
  },
  dashboard: {
    summary: `${API_BASE}/dashboard/summary`,
  },
  excelImports: {
    preview: `${API_BASE}/imports/excel/preview`,
    import: `${API_BASE}/imports/excel/import`,
  },
  settings: {
    company: `${API_BASE}/settings/company`,
    users: `${API_BASE}/settings/users`,
    userById: (userId: string) => `${API_BASE}/settings/users/${userId}`,
    ai: `${API_BASE}/settings/ai`,
  },
  documents: {
    root: `${API_BASE}/documents`,
    process: (documentId: string) => `${API_BASE}/documents/${documentId}/process`,
    reindex: `${API_BASE}/documents/reindex`,
  },
  knowledge: {
    search: `${API_BASE}/knowledge/search`,
  },
  machines: {
    root: `${API_BASE}/machines`,
    byId: (machineId: string) => `${API_BASE}/machines/${machineId}`,
  },
  materials: {
    root: `${API_BASE}/materials`,
    byId: (materialId: string) => `${API_BASE}/materials/${materialId}`,
  },
  products: {
    root: `${API_BASE}/products`,
    byId: (productId: string) => `${API_BASE}/products/${productId}`,
    boms: (productId: string) => `${API_BASE}/products/${productId}/boms`,
    bomById: (productId: string, bomId: string) =>
      `${API_BASE}/products/${productId}/boms/${bomId}`,
    activateBom: (productId: string, bomId: string) =>
      `${API_BASE}/products/${productId}/boms/${bomId}/activate`,
    archiveBom: (productId: string, bomId: string) =>
      `${API_BASE}/products/${productId}/boms/${bomId}/archive`,
    materialRequirements: (productId: string) =>
      `${API_BASE}/products/${productId}/material-requirements`,
    routings: (productId: string) => `${API_BASE}/products/${productId}/routings`,
    routingById: (productId: string, routingId: string) =>
      `${API_BASE}/products/${productId}/routings/${routingId}`,
    activateRouting: (productId: string, routingId: string) =>
      `${API_BASE}/products/${productId}/routings/${routingId}/activate`,
  },
  inventories: {
    root: `${API_BASE}/inventories`,
    transactions: `${API_BASE}/inventories/transactions`,
    receive: `${API_BASE}/inventories/receive`,
    issue: `${API_BASE}/inventories/issue`,
    adjust: `${API_BASE}/inventories/adjust`,
    transfer: `${API_BASE}/inventories/transfer`,
  },
  warehouses: {
    root: `${API_BASE}/warehouses`,
    byId: (warehouseId: string) => `${API_BASE}/warehouses/${warehouseId}`,
  },
  workCenters: {
    root: `${API_BASE}/work-centers`,
    byId: (workCenterId: string) => `${API_BASE}/work-centers/${workCenterId}`,
    deactivate: (workCenterId: string) => `${API_BASE}/work-centers/${workCenterId}/deactivate`,
  },
  productionOrders: {
    root: `${API_BASE}/production-orders`,
    byId: (productionOrderId: string) => `${API_BASE}/production-orders/${productionOrderId}`,
    materialRequirements: (productionOrderId: string) =>
      `${API_BASE}/production-orders/${productionOrderId}/material-requirements`,
    operations: (productionOrderId: string) =>
      `${API_BASE}/production-orders/${productionOrderId}/operations`,
    startOperation: (productionOrderId: string, operationId: string) =>
      `${API_BASE}/production-orders/${productionOrderId}/operations/${operationId}/start`,
    completeOperation: (productionOrderId: string, operationId: string) =>
      `${API_BASE}/production-orders/${productionOrderId}/operations/${operationId}/complete`,
  },
} as const;

File: src/FactoryMind.Api/Routing/ApiRoutes.cs
namespace FactoryMind.Api.Routing;

public static class ApiRoutes {
    public const string Base = "/api";
    public const string Health = "/health";

    public static class Auth {
        public const string Group = Base + "/auth";
        public const string Login = "/login";
        public const string Refresh = "/refresh";
        public const string Logout = "/logout";
    }

    public static class Conversations {
        public const string Group = Base + "/conversations";
        public const string Root = "";
        public const string Messages = "/{conversationId:guid}/messages";
        public const string StreamMessage = Messages + "/stream";
    }

    public static class Documents {
        public const string Group = Base + "/documents";
        public const string Root = "";
        public const string Process = "/{documentId:guid}/process";
        public const string Reindex = "/reindex";
    }

    public static class Dashboard {
        public const string Group = Base + "/dashboard";
        public const string Summary = "/summary";
    }

    public static class ExcelImports {
        public const string Group = Base + "/imports/excel";
        public const string Preview = "/preview";
        public const string Import = "/import";
    }

    public static class Settings {
        public const string Group = Base + "/settings";
        public const string Company = "/company";
        public const string Users = "/users";
        public const string UserById = Users + "/{userId:guid}";
        public const string Ai = "/ai";
    }

    public static class Knowledge {
        public const string Group = Base + "/knowledge";
        public const string Search = "/search";
    }

    public static class Machines {
        public const string Group = Base + "/machines";
        public const string Root = "";
        public const string ById = "/{machineId:guid}";
    }

    public static class Materials {
        public const string Group = Base + "/materials";
        public const string Root = "";
        public const string ById = "/{materialId:guid}";
    }

    public static class Products {
        public const string Group = Base + "/products";
        public const string Root = "";
        public const string ById = "/{productId:guid}";
        public const string Boms = ById + "/boms";
        public const string BomById = Boms + "/{bomId:guid}";
        public const string ActivateBom = BomById + "/activate";
        public const string ArchiveBom = BomById + "/archive";
        public const string MaterialRequirements = ById + "/material-requirements";
        public const string Routings = ById + "/routings";
        public const string RoutingById = Routings + "/{routingId:guid}";
        public const string ActivateRouting = RoutingById + "/activate";
    }

    public static class Inventories {
        public const string Group = Base + "/inventories";
        public const string Root = "";
        public const string Transactions = "/transactions";
        public const string Receive = "/receive";
        public const string Issue = "/issue";
        public const string Adjust = "/adjust";
        public const string Transfer = "/transfer";
    }

    public static class ProductInventories {
        public const string Group = Base + "/product-inventories";
        public const string Root = "";
        public const string Transactions = "/transactions";
    }

    public static class Warehouses {
        public const string Group = Base + "/warehouses";
        public const string Root = "";
        public const string ById = "/{warehouseId:guid}";
    }

    public static class WorkCenters {
        public const string Group = Base + "/work-centers";
        public const string Root = "";
        public const string ById = "/{workCenterId:guid}";
        public const string Deactivate = ById + "/deactivate";
    }

    public static class ProductionOrders {
        public const string Group = Base + "/production-orders";
        public const string Root = "";
        public const string ById = "/{productionOrderId:guid}";
        public const string Release = ById + "/release";
        public const string Start = ById + "/start";
        public const string Complete = ById + "/complete";
        public const string Cancel = ById + "/cancel";
        public const string MaterialRequirements = ById + "/material-requirements";
        public const string Operations = ById + "/operations";
        public const string OperationById = Operations + "/{operationId:guid}";
        public const string StartOperation = OperationById + "/start";
        public const string CompleteOperation = OperationById + "/complete";
    }
}
